### ISSUE TITLE
Suggestion: Don't delete the About folders/xml files when recursively deleting/batch redownloading mods

### ISSUE THREAD

**Issue by Ookaishi** [RimSort/RimSort#380]
**Is your feature request related to a problem? Please describe.**
I have recently swapped to RimSort and I wanted to do a full redownload of all of my mods, sadly the process broke halfway through and now I'm left with a ton of Invalid.Item pseudo-mods which can't be used to redownload the actual mods.
The folders, mod steam workshop IDs are all there, there's just no way to simply attempt a redownload using that.
Having a backup of the modlist.xml doesn't help in this case.
**Describe the solution you'd like**
1) Not deleting the About folders/xml files as to allow proper identification of mods in case something happens.
2) 2-3 Backups (not temp files) of the steamcmd.txt files used to batch download mods, that can be re-used in case of crashes.
3) Have the mods that are being redownloaded placed in a temporary location until they're validated and then moved over to the final location (Although this is a slower option and not always feasible, it's worth considering nonetheless)
**Describe alternatives you've considered**
An alternative would be allowing batch downloading using the modlist.xml, similar to the way RimPy handles missing mods.
Also a backup of the current/active mod list and inactive mods list would be useful as it can be used to properly track mods in situations such as this.
Another alternative would be caching the About folders of every mod somewhere else, if splitting those and the mod data into two different folders and then "merging" them back through sym/hardlinking isn't possible.
**Additional context**
A bit of extra info: Trying to redownload the mods manually results in a failed download, usually saying that the mod could not be found or that no publishedfield was provided.
For context, this is what over half of my mods list is atm:
![image](https://github.com/RimSort/RimSort/assets/22888160/aba695c5-34d0-40f8-b0eb-9bc35bfe5c80)

_Edited to add another alternative solution_

**Comment by Ookaishi** [comment]
A little update: Managed to recover some of the mods through RimPy's downloader (ugh...oh well.) and through manually downloading them directly with steamCMD; Since the folders that were left still had the workshop IDs as names, all it took was a batch script to write the commands out to pass to steamCMD in a file and then running that in steamCMD.
If interested I can post the script here.

**Comment by Chunnyluny** [comment]
please do so others can maybe use it if they think it is useful for their own cases! thanks for finding your own solution and share it so others can recover their fieles back if they are originally rimpy users!!

**Comment by Ookaishi** [comment]
Here's the script, it needs to be put in a .bat file, it will generate a file with almost all of the appropriate download commands based on what folders w/ steamIDs you still have left, just point it to your mods folder; Once that's done you just need to append 'login anonymous' at the beginning of the file and 'quit' at the very end;
Also beware that this **will** mark every present folder, including those of mods downloaded from sources other than Steam Workshop/Through SteamCMD, so you **will** need to doublecheck for those.
If anyone has a better implementation of this script, please do post it, as this was quickly thrown together.

`echo on
cd "Path_to_your_Rimworld/mods folder"
setlocal EnableDelayedExpansion 

for /d %%a in (*) do (
set fileVariable=%%a
set command1=workshop_download_item 294100
set command2=validate
echo !command1! !fileVariable! !command2! >> files1.txt
)`

After running, you must _manually_ run SteamCMD through a Commandprompt window with the + "files1.txt" parameter, the "files1.txt" will be created in the mods folder, so I advise you to move it somewhere easier to access, preferably in the same folder as SteamCMD itself.

**Comment by twstagg** [comment]
You can always export your mod list so that you can easily reimport/redownload the mods from it.

Can you elaborate more on the actual issue that occurred? Was this an issue with SteamCMD redownload?

If so, it may be related to #451 for the most part. I am working on improving the interactions between watchdog and the metadata in it's current form to reduce race conditions and make it overall more reliable as well when mods are created/deleted. I have less hassle on that side of things re: SteamCMD, and more so with Steam client, but I am researching options.

**Comment by Ookaishi** [comment]
It is related to #451 , yes. It's basically the same issue, only thing I have noticed so far is that this specific problem never happened to me when using RimPy;
If I were to take a guess, it's a problem with SteamCMD itself, happens randomly in my case;
An example would be me trying to redownload a mod manually (or update it), adding it to the download list and SteamCMD simply complaining about not finding the mod, although it's clearly available; Restarting RimSort and/or checking to see if the original mod's folder is completely gone generally fixes the issue.
I did notice, however, that RimSort won't fully delete a mod, leaving behind its empty folder which apparently throws SteamCMD for a loop.
